src/expr_tree/impl_expr_tree_Expr.jl

```diff
@@ -84,6 +84,8 @@ function _get_expr_children(ex::Expr)
   end
 end
 
+@inline _sum_expr_trees(exprs::Vector) = Expr(:call, :+, exprs...)
```

Comment:
Is there an element type here?

Reply:
I would like to type it, but I must support 
```julia
expr1 = :(x[1])
expr2 = 4
_sum_expr_trees([expr1, expr2])
```
so typing `exprs` is not possible.
Before calling this line, I check that every components or `exprs` are well typed, in that case being an `Expr` or a `Number`.
The same function exists for other tree structures, and these ones are more stricly typed.
In practice, PartiallySeparableNLPModels.jl should not use this one, but I implemented it to keep the module consistent over the several trees supported.